### ISSUE TITLE
Loading API gives 502 bad gateway

### ISSUE THREAD

**Issue by krishnathokala** [api-platform/api-platform#797]
HI All, 

When i load the API using http://localhost:8080/ URL it gives me "502 Bad Gateway". Could anyone please guide me what is the issue and how to fix this.

Thanks in Advance.

**Comment by teohhanhui** [comment]
Did you verify that PHP-FPM in the `php` container is already listening?

`docker-compose logs -f php`

**Comment by vbourdeix** [comment]
Don't know if that will help, but when that happens, I docker-compose stop then docker-compose up -d my different services and generally it works. I don't like the fact that I don't really understand this issue, but it allowed me to continue working on my project without wasting time on Docker related stuff.

**Comment by wuestkamp** [comment]
A fast fix for this is to call `docker-compose exec php php-fpm -D` to start php-fpm again. I did a complete rebuild of docker-compose and still getting this error.

Anyone got an idea why this happens? It does only so on my local and not my Gcloud Kubernetes cluster.

The [Dockerfile](https://github.com/api-platform/api-platform/blob/master/api/Dockerfile) should run php-fpm.

v2.3.1

**Comment by wildcard27** [comment]
Was there any fix for this? I have just upgraded to 2.3.6 and am getting the same problem.

Everytime I restart the container, I need to run `docker-compose exec php php-fpm -D`. This is getting super annoying

**Comment by wildcard27** [comment]
```
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response
php_1          | /tmp:5432 - no response

```
etc, etc, etc

**Comment by wildcard27** [comment]
It seems that the problem is that the `api` container waits for PHP-FPM to be ready in `docker-entrypoint.sh` but is perhaps timing out?

**Comment by wildcard27** [comment]
@aaa2000 Have already done that

**Comment by baddlan** [comment]
There is a good chance `DATABASE_URL` is not set. Try checking your env vars inside the `php` container.

**Comment by wildcard27** [comment]
@baddlan The `docker-compose.yml` doesn't have that param: https://github.com/api-platform/api-platform/blob/v2.3.6/docker-compose.yml

**Comment by baddlan** [comment]
I am actually on v2.3.4 which had that param before it got omitted in v2.3.5.
https://github.com/api-platform/api-platform/blob/a6f5af491918f472799d52bad621d5a10af54d8b/docker-compose.yml#L15-L16

**Comment by natelenart** [comment]
I just downloaded a stock version of v2.3.6 (Linux, if it matters) and hit this exact issue. I'm new to API platform and I assumed this would be a quick install and then unfortunately spent an hour tracking this down. The good news is that adding the env_file directive to docker-compose.yml fixed it, as per @baddlan's suggestion.

**Comment by dunglas** [comment]
In the latest version (master), the .env file is handled directly by Symfony DotEnv, Docker doesn’t manage it anymore. Can you try if you have the same issue with master please?

**Comment by wildcard27** [comment]
This should mark 2.3.6 as broken, then. There will be many people who will have the same issue.

**Comment by edwh** [comment]
Hi, also a newbie to API Platform and hit this.  Starting PHP-FPM works, but the env_file doesn't seem to help me.

In simple terms, how do I switch to using master?

**Comment by piepkrak** [comment]
Updating the images solved this problem for me.

```bash
$ docker-compose pull
$ docker-compose up -d
```

**Comment by edwh** [comment]
Did it solve it permanently, though?  Or did that just happen to restart PHP-FPM once for you?

**Comment by piepkrak** [comment]
@edwh ```docker-compose stop``` followed by ```docker-compose up -d``` proves that updating the containers did NOT work. I updated my former comment. Thank you for the critical note.

**Comment by piepkrak** [comment]
@edwh It solved it permanently. Running ```docker-compose stop``` followed by ```docker-compose up -d``` will have the API running. It is giving me a gateway timeout the first few seconds after startup. Possibly waiting for another service to become available.

**Comment by mok123** [comment]
Hi there,

i install the api due to the documentation.
I can reach the startpage loalhost.
But by clicking any button on thestartpage i get 502 Bad Gateway error!

Here is my log:

> docker-compose logs -f php
Attaching to api_php_1
php_1          | Loading composer repositories with package information
php_1          | Installing dependencies (including require-dev) from lock file
php_1          | 
php_1          | Prefetching 14 packages 🎶 💨
php_1          |   - Downloading (100%)
php_1          | 
php_1          | Package operations: 14 installs, 0 updates, 0 removals
php_1          |   - Installing league/html-to-markdown (4.8.1): Loading from cache
php_1          |   - Installing symfony/stopwatch (v4.2.3): Loading from cache
php_1          |   - Installing symfony/process (v4.2.3): Loading from cache
php_1          |   - Installing symfony/polyfill-php72 (v1.10.0): Loading from cache
php_1          |   - Installing symfony/options-resolver (v4.2.3): Loading from cache
php_1          |   - Installing php-cs-fixer/diff (v1.3.0): Loading from cache
php_1          |   - Installing composer/xdebug-handler (1.3.2): Loading from cache
php_1          |   - Installing composer/semver (1.4.2): Loading from cache
php_1          |   - Installing friendsofphp/php-cs-fixer (v2.14.2): Loading from cache
php_1          |   - Installing easyrdf/easyrdf (0.9.1): Loading from cache
php_1          |   - Installing api-platform/schema-generator (v2.1.0): Loading from cache
php_1          |   - Installing symfony/var-dumper (v4.2.3): Loading from cache
php_1          |   - Installing symfony/web-profiler-bundle (v4.2.3): Loading from cache
php_1          |   - Installing symfony/profiler-pack (v1.0.4): Loading from cache
php_1          | Generating autoload files
php_1          | Executing script cache:clear [OK]
php_1          | Executing script assets:install public [OK]
php_1          | 
php_1          | Waiting for db to be ready...

Can anyone help?

Thx

**Comment by PapsOu** [comment]
> Was there any fix for this? I have just upgraded to 2.3.6 and am getting the same problem.
> 
> Everytime I restart the container, I need to run `docker-compose exec php php-fpm -D`. This is getting super annoying

I've got the same behavior with version 2.4.1

**Comment by teohhanhui** [comment]
@PapsOu Are you able to reproduce the problem with a new project following the instructions here? https://api-platform.com/docs/distribution/#using-the-official-distribution-recommended

(Download the `.tar.gz`, not the `.zip` please.)

**Comment by PapsOu** [comment]
@teohhanhui I was following this doc when the problem occurs (I always download `tar.gz` when it's possible).

**Comment by teohhanhui** [comment]
@PapsOu I've just tried starting clean following the instructions in the docs, but I've been unable to reproduce the issue. If possible, please provide more information that will help us figure this out.

**Comment by shehi** [comment]
In v2.4.2 two containers crash at first spin-up of docker-compose (varnish and nginx). After attempting to restart, nginx starts, but varnish never starts. Regardless, endpoints work now.

In master (3a16bd4b6c095a7928e363237cfd6c73225d2938) all containers are up. But "Bad request" is still there. Logs output has nothing beyond "Waiting for db to be ready..." line:

![image](https://user-images.githubusercontent.com/653569/55890103-68ad3980-5bb2-11e9-9a56-98406096746a.png)

**Comment by wuestkamp** [comment]
If you don't pass `APP_ENV=prod` to the container, then it'll wait until the database is reachable. This logic is in `docker-entrypoint.sh`.

So in my Kubernetes deployment I configured this and the error disappeared.

```
      containers:
      - image: api-image
        imagePullPolicy: Never
        name: api
        resources: {}
        env:
          - name: APP_ENV
            value: prod
```

**Comment by pitrackster** [comment]
Hi there !
Just installed the latest release 2.4.5(.tar.gz) version og api-platform and followed the tutorial... same problem for me: I'm seeing a `502 Bad Gateway` Nginx error when trying to connect to `https://localhost:8443/`... 

Pulling latest images does not solve any issue, starting manually php-fpm `docker-compose exec php php-fpm -D` does make the `Bad Gateway` disappear bu no js nore css are loaded (404)

Any clue on this ?

thanks
![after_running_php_fpm_manually](https://user-images.githubusercontent.com/6024701/61784381-e582bd00-ae09-11e9-8a2d-da0b3c848f8d.png)

**Comment by teohhanhui** [comment]
@pitrackster That looks like a network / server issue. Please try again.

**Comment by pitrackster** [comment]
@teohhanhui I've tried several times already, each time doing a clean install
Here are the steps I follow:
- stop services
- delete containers
- delete images
- `docker-compose pull`
- `docker-compose up -d`
- go to localhost:8443 => Bad Gateway
- `docker-compose exec php php-fpm -D` no more Bad Gateway but no assets

What should I retry especially ?

**Comment by teohhanhui** [comment]
Can you share the output of `docker-compose logs` when you see the `502 Bad Gateway`?

Because from your screenshot (try not to use screenshots for log messages that you could just copy and paste), it seems like that was an error with the response from flex.symfony.com

**Comment by pitrackster** [comment]
```bash
php_1          | "https://flex.symfony.com/versions.json" does not contain valid JSON
php_1          | Parse error on line 1:
php_1          | 
php_1          | ^
php_1          | Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
php_1          | https://flex.symfony.com/versions.json could not be fully loaded, package information was loaded from the local cache and may be out of date
php_1          | Restricting packages listed in "symfony/symfony" to "4.3.*"
php_1          | Loading composer repositories with package information
php_1          | Installing dependencies (including require-dev) from lock file
php_1          | 
php_1          | Prefetching 15 packages 🎶 💨
```

**Comment by teohhanhui** [comment]
Anyway, it seems like that's not a fatal error. You just need to wait until the container's startup is complete.

**Comment by pitrackster** [comment]
I think it get stuck there yes ;-) Every containers are started...
```bash
docker-compose up -d
Creating network "api-platform_default" with the default driver
Creating api-platform_mercure_1 ... done
Creating api-platform_db_1      ... done
Creating api-platform_admin_1       ... done
Creating api-platform_client_1      ... done
Creating api-platform_php_1     ... done
Creating api-platform_api_1     ... done
Creating api-platform_cache-proxy_1 ... done
Creating api-platform_h2-proxy_1    ... done
```

but php-fpm is not started

**Comment by teohhanhui** [comment]
Hmm... Can you confirm that with `docker-compose logs -f php`?

But anyway, it's not a problem with our Docker setup. It looks like a network connectivity  issue. If you're behind a proxy, you might run into such issues.

**Comment by pitrackster** [comment]
```
docker-compose logs -f php  
Attaching to api-platform_php_1
php_1          | "https://flex.symfony.com/versions.json" does not contain valid JSON
php_1          | Parse error on line 1:
php_1          | 
php_1          | ^
php_1          | Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
php_1          | https://flex.symfony.com/versions.json could not be fully loaded, package information was loaded from the local cache and may be out of date
php_1          | Restricting packages listed in "symfony/symfony" to "4.3.*"
php_1          | Loading composer repositories with package information
php_1          | Installing dependencies (including require-dev) from lock file
php_1          | 
php_1          | Prefetching 15 packages 🎶 💨
```
I dont think I am behind a proxy... If I can access the url directly via Firefox why could'nt access it via a Docker command ?

**Comment by teohhanhui** [comment]
Try changing this line:

https://github.com/api-platform/api-platform/blob/8dd71fd2c16e68b040eca58d80a8cd9a280413de/api/docker/php/docker-entrypoint.sh#L21

to this:

```shell
composer install --prefer-dist --no-progress --no-suggest --no-interaction -vvv
```

Then do:

```shell
docker-compose build php
docker-compose up -d
```

Hopefully that will tell us more about what's going on.

**Comment by pitrackster** [comment]
Sorry I thought It would be an easy fix :(

From the first command (rebuild php service)
```bash
docker-compose build php
Building php
Step 1/29 : ARG PHP_VERSION=7.3
Step 2/29 : ARG NGINX_VERSION=1.15
Step 3/29 : ARG VARNISH_VERSION=6.2
Step 4/29 : FROM php:${PHP_VERSION}-fpm-alpine AS api_platform_php
 ---> 5272cea79594
Step 5/29 : RUN apk add --no-cache              acl             file            gettext                 git     ;
 ---> Using cache
 ---> 05b3cabc4b44
Step 6/29 : ARG APCU_VERSION=5.1.17
 ---> Using cache
 ---> 93184c275bea
Step 7/29 : RUN set -eux;       apk add --no-cache --virtual .build-deps                $PHPIZE_DEPS            icu-dev                 libzip-dev              postgresql-dev        zlib-dev         ;               docker-php-ext-configure zip --with-libzip;     docker-php-ext-install -j$(nproc)               intl            pdo_pgsql               zip     ;     pecl install             apcu-${APCU_VERSION}    ;       pecl clear-cache;       docker-php-ext-enable           apcu            opcache         ;               runDeps="$(           scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions                    | tr ',' '\n'                   | sort -u                       | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }'        )";     apk add --no-cache --virtual .api-phpexts-rundeps $runDeps;             apk del .build-deps
 ---> Using cache
 ---> b8c46a7b7a4d
Step 8/29 : COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
 ---> Using cache
 ---> 24adbce067b1
Step 9/29 : RUN ln -s $PHP_INI_DIR/php.ini-production $PHP_INI_DIR/php.ini
 ---> Using cache
 ---> 7588898e88ff
Step 10/29 : COPY docker/php/conf.d/api-platform.ini $PHP_INI_DIR/conf.d/api-platform.ini
 ---> Using cache
 ---> 803e6f14a94b
Step 11/29 : ENV COMPOSER_ALLOW_SUPERUSER=1
 ---> Using cache
 ---> 93e2c95176d1
Step 12/29 : RUN set -eux;      composer global require "symfony/flex" --prefer-dist --no-progress --no-suggest --classmap-authoritative;       composer clear-cache
 ---> Using cache
 ---> ef341c043c7c
Step 13/29 : ENV PATH="${PATH}:/root/.composer/vendor/bin"
 ---> Using cache
 ---> 9dc5e53e6f7b
Step 14/29 : WORKDIR /srv/api
 ---> Using cache
 ---> 0231e3a76386
Step 15/29 : ARG APP_ENV=prod
 ---> Using cache
 ---> 4b076ae67326
Step 16/29 : COPY composer.json composer.lock symfony.lock ./
 ---> Using cache
 ---> 940474935990
Step 17/29 : RUN set -eux;      composer install --prefer-dist --no-dev --no-scripts --no-progress --no-suggest;        composer clear-cache
 ---> Using cache
 ---> 8efa473898b3
Step 18/29 : COPY .env ./
 ---> Using cache
 ---> 25646ceadcca
Step 19/29 : RUN composer dump-env prod;        rm .env
 ---> Using cache
 ---> 659ce564ca84
Step 20/29 : COPY bin bin/
 ---> Using cache
 ---> 8967b1a9b1e5
Step 21/29 : COPY config config/
 ---> Using cache
 ---> fb8ba186d3be
Step 22/29 : COPY public public/
 ---> Using cache
 ---> bbe5c275b1e9
Step 23/29 : COPY src src/
 ---> Using cache
 ---> fe16a90fc02d
Step 24/29 : RUN set -eux;      mkdir -p var/cache var/log;     composer dump-autoload --classmap-authoritative --no-dev;       composer run-script --no-dev post-install-cmd;  chmod +x bin/console; sync
 ---> Using cache
 ---> 01d5c399d4f2
Step 25/29 : VOLUME /srv/api/var
 ---> Using cache
 ---> 9f0fca148aa2
Step 26/29 : COPY docker/php/docker-entrypoint.sh /usr/local/bin/docker-entrypoint
 ---> f91217b26fbd
Step 27/29 : RUN chmod +x /usr/local/bin/docker-entrypoint
 ---> Running in 06413fdc2f00
Removing intermediate container 06413fdc2f00
 ---> bb1842e0447f
Step 28/29 : ENTRYPOINT ["docker-entrypoint"]
 ---> Running in d4ffec6ff3d5
Removing intermediate container d4ffec6ff3d5
 ---> b351375e1e47
Step 29/29 : CMD ["php-fpm"]
 ---> Running in 2a4af9df0e4d
Removing intermediate container 2a4af9df0e4d
 ---> d8931f4db99b

Successfully built d8931f4db99b
Successfully tagged quay.io/api-platform/php:latest
```

Then

```bash
docker-compose up -d
api-platform_admin_1 is up-to-date
api-platform_client_1 is up-to-date
api-platform_mercure_1 is up-to-date
api-platform_db_1 is up-to-date
Recreating api-platform_php_1 ... done
Recreating api-platform_api_1 ... done
Recreating api-platform_cache-proxy_1 ... done
Recreating api-platform_h2-proxy_1    ... done
```


```bash
docker-compose logs -f php       
Attaching to api-platform_php_1
php_1          | Reading ./composer.json
php_1          | Loading config file ./composer.json
php_1          | Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
php_1          | Executing command (/srv/api): git branch --no-color --no-abbrev -v
php_1          | Executing command (/srv/api): git describe --exact-match --tags
php_1          | Executing command (/srv/api): git log --pretty="%H" -n1 HEAD
php_1          | Executing command (/srv/api): hg branch
php_1          | Executing command (/srv/api): fossil branch list
php_1          | Executing command (/srv/api): fossil tag list
php_1          | Executing command (/srv/api): svn info --xml
php_1          | Reading /root/.composer/composer.json
php_1          | Loading config file /root/.composer/composer.json
php_1          | Reading /srv/api/vendor/composer/installed.json
php_1          | Reading /root/.composer/vendor/composer/installed.json
php_1          | Loading plugin Symfony\Flex\Flex
php_1          | Reading /root/.composer/cache/repo/https---flex.symfony.com/versions.json from cache
php_1          | Downloading https://flex.symfony.com/versions.json
php_1          | Downloading https://flex.symfony.com/versions.json
php_1          | Downloading https://flex.symfony.com/versions.json
php_1          | "https://flex.symfony.com/versions.json" does not contain valid JSON
php_1          | Parse error on line 1:
php_1          | 
php_1          | ^
php_1          | Expected one of: 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '['
php_1          | https://flex.symfony.com/versions.json could not be fully loaded, package information was loaded from the local cache and may be out of date
php_1          | Restricting packages listed in "symfony/symfony" to "4.3.*"
php_1          | Running 1.8.6 (2019-06-11 15:03:05) with PHP 7.3.7 on Linux / 5.0.0-20-generic
php_1          | Reading ./composer.lock
php_1          | Loading composer repositories with package information
php_1          | Installing dependencies (including require-dev) from lock file
php_1          | Reading ./composer.lock
php_1          | Resolving dependencies through SAT
php_1          | Looking at all rules.
php_1          | 
php_1          | Dependency resolution completed in 0.002 seconds
php_1          | 
php_1          | Prefetching 15 packages 🎶 💨
php_1          |   - DownloadingDownloading https://codeload.github.com/thephpleague/html-to-markdown/legacy.zip/250d1bf45f80d15594fb6b316df777d6d4c97ad1
php_1          | Downloading https://codeload.github.com/symfony/stopwatch/legacy.zip/6b100e9309e8979cf1978ac1778eb155c1f7d93b
php_1          | Downloading https://codeload.github.com/symfony/process/legacy.zip/856d35814cf287480465bb7a6c413bb7f5f5e69c
php_1          | Downloading https://codeload.github.com/symfony/options-resolver/legacy.zip/914e0edcb7cd0c9f494bc023b1d47534f4542332
php_1          | Downloading https://codeload.github.com/PHP-CS-Fixer/diff/legacy.zip/78bb099e9c16361126c86ce82ec4405ebab8e756
php_1          | Downloading https://codeload.github.com/composer/xdebug-handler/legacy.zip/46867cbf8ca9fb8d60c506895449eb799db1184f
php_1          | Downloading https://codeload.github.com/composer/semver/legacy.zip/46d9139568ccb8d9e7cdd4539cab7347568a5e2e
php_1          | Downloading https://codeload.github.com/FriendsOfPHP/PHP-CS-Fixer/legacy.zip/20064511ab796593a3990669eff5f5b535001f7c
php_1          | Downloading https://codeload.github.com/njh/easyrdf/legacy.zip/acd09dfe0555fbcfa254291e433c45fdd4652566
php_1          | Downloading https://codeload.github.com/api-platform/schema-generator/legacy.zip/849d34cc19fd86851d3015a8cbd3d8fe00da3ef8
php_1          | Downloading https://codeload.github.com/nikic/PHP-Parser/legacy.zip/1bd73cc04c3843ad8d6b0bfc0956026a151fc420
php_1          | Downloading https://codeload.github.com/symfony/maker-bundle/legacy.zip/d262c2cace4d9bca99137a84f6fc6ba909a17e02
php_1          | Downloading https://codeload.github.com/symfony/var-dumper/legacy.zip/f974f448154928d2b5fb7c412bd23b81d063f34b
php_1          | Downloading https://codeload.github.com/symfony/web-profiler-bundle/legacy.zip/ca3a3c8558bc641df7c8c2c546381ccd78d0777a
php_1          | Downloading https://codeload.github.com/symfony/profiler-pack/legacy.zip/99c4370632c2a59bb0444852f92140074ef02209

```

And nothing else when pressing F5 serveral times...

I also tried to install symfony/flex from inside another docker service and it works perfectly... 

```
docker-compose exec web /bin/bash
root@669a8bf34e5f:/var/www# composer require symfony/flex
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Using version ^1.4 for symfony/flex
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "4.3.*"
Package operations: 0 installs, 1 update, 0 removals
  - Updating symfony/flex (v1.4.4 => v1.4.5): As there is no 'unzip' command installed zip files are being unpacked using the PHP zip extension.
This may cause invalid reports of corrupted archives. Besides, any UNIX permissions (e.g. executable) defined in the archives will be lost.
Installing 'unzip' may remediate them.
Downloading (100%)         
Writing lock file
Generating autoload files
ocramius/package-versions:  Generating version class...
ocramius/package-versions: ...done generating version class
Executing script cache:clear [OK]
Executing script assets:install public [OK]
```

**Comment by teohhanhui** [comment]
That's strange. I'll see if I can reproduce the problem.

**Comment by teohhanhui** [comment]
Nope. I'm absolutely unable to reproduce the issue: https://gist.github.com/teohhanhui/ede9d16f7b85ed67523f8c7eb7e607c5

**Comment by pitrackster** [comment]
:( OK thanks anyway I'll try to figure out what is going on restarting from scratch...

**Comment by jberry93** [comment]
I ran into this same problem and found out after running `docker-compose exec php bin/console` I got this error:

```
OCI runtime exec failed: exec failed: container_linux.go:345: starting container process caused "exec: \"bin/console\": permission denied": unknown
```
What fixed the 502 bad gateway problem for me was this [stackoverflow answer](https://stackoverflow.com/a/51370758). Based on that answer, I ran the following to grant permission:

```
docker-compose exec php chmod 755 bin/console
```

I also restarted my containers (`docker-compose restart`) just to be safe and it looks like it remedied the problem. This may not be a sure-fire or even correct way to resolve the issue but it worked for me. Hope this helps!

**Comment by MLDMoritz** [comment]
This line is your bummer - your app environment is not dev and thus it won't dump your assets.

```
if [ "$APP_ENV" != 'prod' ]; then
    composer install --prefer-dist --no-progress --no-suggest --no-interaction
fi
```

**Comment by jessequinn** [comment]
@jberry93 your solution worked for me. No need to `docker-compose exec php php-fpm -D`. Using master as of yesterday.

**Comment by jwmickey** [comment]
This may not be relevant to the thread, but I ended up here after a search and realized that I had changed the password for postgres in docker-compose.yml but forgot to add a .env.local file in api/ to override the default !ChangeMe! password.  The php container logs showed it was stuck at "Waiting for db to be ready".  Once I changed the password, db migrations were applied and then it was able to respond to requests.

**Comment by Ousret** [comment]
If anyone is stuck on this nowadays.. 

Juste patch the following file `api/docker/php/docker-entrypoint.sh`.

```sh
        echo "Waiting for db to be ready..."
	until php bin/console doctrine:query:sql "SELECT 1" > /dev/null 2>&1; do
		sleep 1
	done

	if ls -A src/Migrations/*.php > /dev/null 2>&1; then
		php bin/console doctrine:migrations:migrate --no-interaction
	fi
```

Just add `php` before `bin/console`. Don't know why but the PHP container does not recognize the following shebang `#!/usr/bin/env php`.

**Comment by MarlonAEC** [comment]
Well ... I'm still stuck :( I'm still getting Bad Gateway and I've already tried everything mentioned above, any ideas ??

**Comment by Surf-N-Code** [comment]
I also had this issue. However, only when I had enabled xdebug via the docker-compose.override.yml
Restarting php-fpm worked once!

Then I checked the php logs when it happened again. It turned out I had two issues:

1. The database url in my env.local was wrong. Therefore the container was stuck waiting for the db
2. I had a migration that caused trouble

Seems to work now after having fixed those issues.
